File: Day014_2025/Q014_GFG.py
class Solution:
    # Function to find equilibrium point in the array.
    def findEquilibrium(self, arr):
        # Q. Equilibrium Point
        total = sum(arr)
        pre_sum = 0

        for i in range(len(arr)):
            pre_sum += arr[i]
            if pre_sum - arr[i] == total - pre_sum:
                return i
        return -1


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        sol = Solution()
        print(sol.findEquilibrium(arr))
        print("~")

File: Day014_2025/Q014_GFG.java
import java.util.*;

class Q014_GFG {
    // Function to find equilibrium point in the array.
    public static int findEquilibrium(int[] arr) {
        // Q. Equilibrium Point
        int total = Arrays.stream(arr).sum();
        int preSum = 0;

        for (int i = 0; i < arr.length; ++i) {
            preSum += arr[i];
            if (preSum - arr[i] == total - preSum) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        sc.nextLine(); // Consume leftover newline

        while (t-- > 0) {
            String[] input = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(input).mapToInt(Integer::parseInt).toArray();

            System.out.println(findEquilibrium(arr));
            System.out.println("~");
        }
        sc.close();
    }
}
